### PR TITLE
Add consumers-wait-for-messages warning

### PR DESCRIPTION
**Pull request by nuzil**
## Purpose of this pull request

This pull request (PR) add a warning for using of consumers-wait-for-messages option. For do not confuse devs.

## Affected DevDocs pages

https://devdocs.magento.com/guides/v2.4/extension-dev-guide/message-queues/config-mq.html?itm_source=devdocs&itm_medium=quick_search&itm_campaign=federated_search&itm_term=

### CODE REVIEW

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md [hunk 1]
```diff
@@ -116,6 +116,9 @@ The [`consumers-wait-for-messages`]({{page.baseurl}}/install-gde/install/cli/ins
 The problem is that every time the cron job `cron_consumers_runner` runs, it spawns a new consumer process, the consumer checks if messages are available, and it terminates itself if there are no messages.
 Meanwhile, the `onlySpawnWhenMessageAvailable` attribute first checks if there are available messages, and it spawns a new consumer process only if there are messages. It means that it does not spawn unneeded processes which take up memory, live for a very short period, and then disappear.
 
+{:.bs-callout-warning}
+Be aware, that [`consumers-wait-for-messages`]({{page.baseurl}}/install-gde/install/cli/install-cli-subcommands-consumers.html) option is a global option and cannot be configured separately for each consumer like `onlySpawnWhenMessageAvailable` option.
```
Comment (BarnyShergold) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:120]:
```suggestion
Be aware, that the [`consumers-wait-for-messages`]({{page.baseurl}}/install-gde/install/cli/install-cli-subcommands-consumers.html) option is a global option and cannot be configured separately for each consumer, like the `onlySpawnWhenMessageAvailable` option.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]

**Review by BarnyShergold — APPROVED** [review]
Nice up date

**Review by dobooth — APPROVED** [review]

**Comment by dobooth** [comment]
@nuzil Just want to confirm that this is a v2.4 issue only. Do we need to update 2.3 as well?

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md [hunk 2]
```diff
@@ -116,6 +116,9 @@ The [`consumers-wait-for-messages`]({{page.baseurl}}/install-gde/install/cli/ins
 The problem is that every time the cron job `cron_consumers_runner` runs, it spawns a new consumer process, the consumer checks if messages are available, and it terminates itself if there are no messages.
 Meanwhile, the `onlySpawnWhenMessageAvailable` attribute first checks if there are available messages, and it spawns a new consumer process only if there are messages. It means that it does not spawn unneeded processes which take up memory, live for a very short period, and then disappear.
 
+{:.bs-callout-warning}
+The [`consumers-wait-for-messages`]({{page.baseurl}}/install-gde/install/cli/install-cli-subcommands-consumers.html) option is a global option and cannot be configured separately for each consumer, such as the `onlySpawnWhenMessageAvailable` option.
```
Comment (keharper) [inline src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:120]: The "Configure consumer behavior" topic linked above should also be updated to note that the --consumers-wait-for-messages option is global.

**Comment by nuzil** [comment]
> @nuzil Just want to confirm that this is a v2.4 issue only. Do we need to update 2.3 as well?

Yes, correct.

File: src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md (hunk 2 above)
Comment (nuzil) [reply src/guides/v2.4/extension-dev-guide/message-queues/config-mq.md:120]: Good call. Made it

**Comment by magento-engcom-team** [comment]
@dobooth the pull request successfully imported.

**Pull request merged**